e2e/cli-plugins/flags_test.go

```diff
@@ -17,3 +19,82 @@ func TestRunGoodArgument(t *testing.T) {
 		Out:      "Hello Cleveland!",
 	})
 }
+
+// TestClashWithGlobalArgs ensures correct behaviour when a plugin
+// has an argument with the same name as one of the globals.
+func TestClashWithGlobalArgs(t *testing.T) {
+	run, _, cleanup := prepare(t)
+	defer cleanup()
+
+	for _, tc := range []struct {
+		name                     string
+		args                     []string
+		expectedOut, expectedErr string
+	}{
+		{
+			name:        "short-without-val",
+			args:        []string{"-D"},
+			expectedOut: "Hello World!",
+			expectedErr: "Plugin debug mode enabled",
+		},
+		{
+			name:        "long-without-val",
+			args:        []string{"--debug"},
+			expectedOut: "Hello World!",
+			expectedErr: "Plugin debug mode enabled",
+		},
+		{
+			name:        "short-with-val",
+			args:        []string{"-c", "Christmas"},
+			expectedOut: "Merry Christmas!",
+			expectedErr: "",
+		},
+		{
+			name:        "short-with-val",
+			args:        []string{"--context", "Christmas"},
+			expectedOut: "Merry Christmas!",
+			expectedErr: "",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			args := append([]string{"helloworld"}, tc.args...)
+			res := icmd.RunCmd(run(args...))
+			res.Assert(t, icmd.Expected{
+				ExitCode: 0,
+				Out:      tc.expectedOut,
+				Err:      tc.expectedErr,
+			})
+		})
+	}
+}
+
+// TestUnknownGlobal checks that unknown globals report errors
+func TestUnknownGlobal(t *testing.T) {
+	run, _, cleanup := prepare(t)
+	defer cleanup()
+
+	t.Run("no-val", func(t *testing.T) {
```

Comment:
Perhaps;

```go
// TestUnknownGlobal checks that unknown globals report errors
func TestUnknownGlobal(t *testing.T) {
	run, _, cleanup := prepare(t)
	defer cleanup()

	for name, args := range map[string][]string{
		"no-val":       {"--unknown", "helloworld"},
		"separate-val": {"--unknown", "foo", "helloworld"},
		"joined-val":   {"--unknown=foo", "helloworld"},
	} {
		t.Run(name, func(t *testing.T) {
			res := icmd.RunCmd(run(args...))
			res.Assert(t, icmd.Expected{
				ExitCode: 125,
				Out:      "",
				Err:      "unknown flag: --unknown",
			})
		})
	}
}
```

Reply:
Ack, will do this. NB `Out: ""` needs to be `Out: icmd.None` to actually check it is empty, `""` means "no check". I only just discovered this, I had been doing it manually everywhere.

Reply:
nice find; didn't know that; that's a bit confusing 😕